### PR TITLE
[ID-741]Fix Case-Insensitive Sorting

### LINKED ISSUE #741: [BUG] Fix Case-Insensitive Sorting
**Describe the bug**
Fix Case-Insensitive Sorting

### PR DESCRIPTION
**Pull request by stefanvit**
## Description
Fix Case-Insensitive Sorting

**Resolves #741 
## Review Time Estimate
- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
- [x ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [x] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by petyos — CHANGES_REQUESTED** [review]
It does not fix sorting as sorting is applied here:

Stephen said, can you add a case insensitivity flag to the Core BB, applied to the Mongo query. He passed me this:

Directory Sort that needs case insensitivity applied https://github.com/rokwire/core-building-block/blob/01e7e72219eb3e43c93520fda6f60497ac25b655/driven/storage/adapter.go#L1544

File: driven/storage/adapter.go [hunk 1]
```diff
@@ -1582,7 +1617,7 @@ func (sa *Adapter) FindPublicAccounts(context TransactionContext, appID string,
 	publicAccounts := make([]model.PublicAccount, 0)
 	for _, item := range results {
 		account := accountFromStorage(item, &appOrg.ID, allAppsOrgs)
-		publicAccount, err := account.GetPublicAccount(false) //TODO: get actual connection status
```
Comment (petyos) [inline driven/storage/adapter.go:1585]: Do not remove this?

File: driven/storage/adapter.go [hunk 2]
```diff
@@ -1540,8 +1557,17 @@ func (sa *Adapter) FindPublicAccounts(context TransactionContext, appID string,
 		pipeline = append(pipeline, bson.M{"$match": bson.M{"profile.unstructured_properties." + k: v}})
 	}
 
-	pipeline = append(pipeline, bson.M{"$match": bson.M{"org_apps_memberships.app_org_id": appOrg.ID, "privacy.public": true}})
```
Comment (petyos) [inline driven/storage/adapter.go:1543]: Why $match is removed?

**Review by petyos — CHANGES_REQUESTED** [review]

**Comment by stefanvit** [comment]
Hi @petyos 
Normalization ($addFields):
Before sorting, the fields profile.last_name and profile.first_name are temporarily converted to lowercase and stored as normalized_last_name and normalized_first_name.
Sorting ($sort):

The documents are then sorted using the normalized fields.
The original fields (profile.last_name and profile.first_name) remain unchanged in the documents.
Pipeline Progression:

These normalized fields exist only temporarily in the pipeline and won't appear in the final output unless explicitly included in a $project stage.

**Review by petyos — APPROVED** [review]
LGTM

**Pull request merged**